src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -51,72 +53,115 @@ impl AccessChecker for PrivilegeAccess {
+                        vec![UserPrivilegeType::Select],
+                    )
+                    .await?
             }
-            Plan::UseDatabase(_) => {}
 
             // Table.
-            Plan::ShowCreateTable(_) => {}
-            Plan::DescribeTable(_) => {}
+            Plan::ShowCreateTable(plan) => {
+                session
+                    .validate_privilege(
+                        &GrantObject::Table(
+                            plan.catalog.clone(),
+                            plan.database.clone(),
+                            plan.table.clone(),
+                        ),
+                        vec![UserPrivilegeType::Select],
+                    )
+                    .await?
+            }
+            Plan::DescribeTable(plan) => {
+                session
+                    .validate_privilege(
+                        &GrantObject::Table(
+                            plan.catalog.clone(),
+                            plan.database.clone(),
+                            plan.table.clone(),
+                        ),
+                        vec![UserPrivilegeType::Select],
+                    )
+                    .await?
+            }
             Plan::CreateTable(plan) => {
                 session
                     .validate_privilege(
                         &GrantObject::Database(plan.catalog.clone(), plan.database.clone()),
-                        UserPrivilegeType::Create,
+                        vec![UserPrivilegeType::Create],
                     )
                     .await?;
             }
             Plan::DropTable(plan) => {
                 session
                     .validate_privilege(
                         &GrantObject::Database(plan.catalog.clone(), plan.database.clone()),
-                        UserPrivilegeType::Drop,
+                        vec![UserPrivilegeType::Drop],
                     )
                     .await?;
             }
             Plan::UndropTable(plan) => {
                 session
                     .validate_privilege(
                         &GrantObject::Database(plan.catalog.clone(), plan.database.clone()),
-                        UserPrivilegeType::Drop,
+                        vec![UserPrivilegeType::Drop],
+                    )
+                    .await?;
+            }
+            Plan::RenameTable(plan) => {
+                // You must have ALTER and DROP privileges for the original table,
+                // and CREATE and INSERT privileges for the new table.
+                session
+                    .validate_privilege(
+                        &GrantObject::Table(
+                            plan.catalog.clone(),
+                            plan.database.clone(),
+                            plan.table.clone(),
+                        ),
+                        vec![UserPrivilegeType::Alter, UserPrivilegeType::Drop],
+                    )
```

Comment:
@BohuTANG  I think you mean like this?

this case :

alter table old_tb_name rename to new_tb_name;

need alter and drop on old_tb_name;

need create and insert on new_tb_name

Reply:
Right!

For example, 
https://github.com/datafuselabs/databend/pull/10191/files#diff-1c7ca9f866afb39e388942d88ae7da6c2167c9e534134d9ff26b1e7353f7ccceR220-R242

Here how can we do?